Sources/ComposableArchitecture/Store.swift

```diff
@@ -697,116 +813,6 @@ public final class Store<State, Action> {
 /// ```
 public typealias StoreOf<R: Reducer> = Store<R.State, R.Action>
 
-extension Reducer {
-  fileprivate func rescope<ChildState, ChildAction>(
```

Comment:
Since stores are now being cached, we think this `rescope` machinery is no longer necessary in most cases. I also did not measure much of a performance difference in its removal in our very limited suite.

@iampatbrown Let me know your thoughts on this, especially if you have apprehensions. I think we could potentially keep `rescope` for stores that are not cached, though I'm not 100% sure what that would look like.

Reply:
No apprehensions :) I'm a fan of the cached approach. 
Scoping via a key path is probably preferable. Computed properties might be an alternative for people using a closure as well.